docs/configuration/tools/router_check.rst

```diff
@@ -0,0 +1,109 @@
+.. _config_tools_router_check_tool:
+
+Route table check tool
+======================
+
+**NOTE: The following configuration is for the router table check tool only and is not part of the Envoy binary.
+The route table check tool is used for testing purposes only.**
+
+Input for the route table check tool. The route table check tool checks if the route returned
+by a router matches what is expected. The tool can be used to check cluster name, virtual cluster name,
+virtual host name, path rewrite, host rewrite, and path redirect matches. Extensions for other
+test cases can be added. The "check" field specifies the expected values in each test case. At least one test
```

Comment:
nit: s/host rewrite/manual and automatic host rewrite/
(and while doing so, please add a test case for automatic host rewrite. We use this flag in Istio to link K8S clusters to external services).